lib/resty/dns/resolver.lua

```diff
@@ -398,7 +414,10 @@ local function parse_response(buf, id)
         answers.errstr = resolver_errstrs[code] or "unknown"
     end
 
-    for i = 1, nan do
+    local nan_nns = nan + nns
+    local nan_nar = nan_nns + nar
+
+    for i = 1, nan_nar do
```

Comment:
Yeah, it is, that means I should just pass the options to the parsing function right?